client/service/src/chain_ops.rs

```diff
@@ -175,25 +336,17 @@ impl<
 					"Stopping after #{} blocks because of an error",
 					link.imported_blocks,
 				);
+				// We've encountered an error, we can stop here.
 				return std::task::Poll::Ready(Ok(()));
 			}
 
-			if link.imported_blocks / 1000 != blocks_before / 1000 {
-				info!(
-					"#{} blocks were imported (#{} left)",
-					link.imported_blocks,
-					count - link.imported_blocks
-				);
-			}
+			let read_block_count = block_stream.read_block_count();
+			let count = block_stream.count();
+			log_importing_status_updates(count, read_block_count, blocks_before, link.imported_blocks);
```

Comment:
```suggestion
			log_importing_status_updates(&block_stream, blocks_before, link.imported_blocks);
```

Would make it shorter.